bootstrap-app/src/cmr/bootstrap/data/fingerprint.clj

```diff
@@ -0,0 +1,101 @@
+(ns cmr.bootstrap.data.fingerprint
+  "Functions to support updating variable fingerprint."
+  (:require
+    [clojure.core.async :as ca :refer [<!!]]
+    [cmr.bootstrap.embedded-system-helper :as helper]
+    [cmr.common.concepts :as concepts]
+    [cmr.common.log :refer (debug info warn error)]
+    [cmr.common.services.errors :as errors]
+    [cmr.metadata-db.data.concepts :as db]
+    [cmr.umm-spec.fingerprint-util :as fingerprint-util]))
+
+(defn- fingerprint-variable
+  "Update the fingerprint of the given variable if necessary."
+  [db provider variable]
+  (let [{:keys [concept-id revision-id deleted metadata]} variable]
+    (when (not deleted)
+      (let [old-fingerprint (get-in variable [:extra-fields :fingerprint])
+            new-fingerprint (fingerprint-util/get-variable-fingerprint metadata)]
+        (when (not= old-fingerprint new-fingerprint)
+          (db/save-concept db
+                           provider
+                           (-> variable
+                               (assoc :revision-id (inc revision-id))
+                               (assoc-in [:extra-fields :fingerprint] new-fingerprint)
+                               (dissoc :revision-date)))
+          (info (format "Updated fingerprint for concept-id: %s" concept-id)))))))
+
+(defn- fingerprint-variable-batch
+  "Update the fingerprints of variables in the given variable batch if necessary."
+  [db provider variable-batch]
+  (dorun
+   (map #(fingerprint-variable db provider %) variable-batch)))
+
+(def ^:private find-variables-sql-part1
+  "Defines the beginning part of the string to construct find variables sql statement."
+  (str "select a.* from metadata_db.cmr_variables a, "
+       "(select concept_id, max(revision_id) rid from metadata_db.cmr_variables "))
+
+(def ^:private find-variables-sql-part2
+  "Defines the ending part of the string to construct find variables sql statement."
+  (str "group by concept_id) b"
+       " where a.concept_id = b.concept_id and a.revision_id = b.rid and a.deleted = 0"))
+
+(defn- find-variables-sql
+  "Returns the sql statement to find the latest variable concept revisions that are not deleted."
+  ([]
+   (str find-variables-sql-part1 find-variables-sql-part2))
+  ([provider-id]
+   (format "%s where provider_id='%s' %s"
```

Comment:
This sees dangerous. Shouldn't we parameterize `provider-id` instead?

Reply:
See above. The `provider-id` has to be a validate id that map to an existing provider in CMR before it can get here.